### PR TITLE
fix crash enabling PFL after deleting EffectState while PFL is disabled

### PR DESCRIPTION
**Pull request by Be-ing**
fixing https://bugs.launchpad.net/mixxx/+bug/1741213

### CODE REVIEW

**Comment by Be-ing** [comment]
Ping. Can we prioritize this critical bug fix?

File: src/effects/defs.h [hunk 1]
```diff
@@ -28,6 +28,11 @@ enum class EffectChainInsertionType {
 
 const int kNumEffectsPerUnit = 4;
 
+// NOTE: Setting this to true will enable string manipulation and calls to
+// qDebug() in the audio engine thread. That may cause audio dropouts, so only
+// enable this when debugging the effects system.
+const bool kEffectDebugOutput = false;
```
Comment (uklotzde) [inline src/effects/defs.h]: A constexpr might be more suitable to enforce evaluation (and optimization) at compile time

**Comment by uklotzde** [comment]
The crash is fixed, but replaced by another minor bug: When re-routing the deck to the effect unit with the effect still activated, it is not applied. You need to deactivate and activate the effect once, before it gets applied again.

**Comment by uklotzde** [comment]
LGTM. Just waiting for the builds to finish.

**Pull request merged**